collector/avro_record.go

```diff
@@ -58,12 +58,51 @@ func (ar avroRecord) extract(pmm *producers.MetricsMessage) error {
 		datapoints := []producers.Datapoint{}
 		for _, field := range ar {
 			fwColLog.Debugf("Adding datapoint %s", field)
+
 			dp := producers.Datapoint{
 				Name:      fmt.Sprintf("%v", field.Fields[0].Datum),
-				Value:     fmt.Sprintf("%v", field.Fields[1].Datum),
 				Unit:      fmt.Sprintf("%v", field.Fields[2].Datum),
 				Timestamp: fmt.Sprintf("%v", time.Now()),
 			}
+
+			value := field.Fields[1].Datum
+			switch value.(type) {
```

Comment:
this looks a little bit strange, why do you switch type if in any case you just assign `value` to `dp.Value`? All this switch is not needed. You can just `dp.Value = value` it will be the same. I think you are missing something here?